src/benchmark/tpch_data_micro_benchmark.cpp

```diff
@@ -0,0 +1,208 @@
+#include "micro_benchmark_basic_fixture.hpp"
+
+#include "expression/expression_functional.hpp"
+#include "operators/join_hash.hpp"
+#include "operators/join_sort_merge.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/table_wrapper.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/encoding_type.hpp"
+#include "storage/storage_manager.hpp"
+#include "tpch/tpch_db_generator.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+class TableWrapper;
+
+// Defining the base fixture class
+class TPCHDataMicroBenchmarkFixture : public MicroBenchmarkBasicFixture {
+ public:
+  void SetUp(::benchmark::State& state) {
+    auto& sm = StorageManager::get();
+    const auto scale_factor = 0.001f;
+
+    if (!sm.has_table("lineitem")) {
+      std::cout << "Generating TPC-H data with scale factor " << scale_factor << " ... " << std::flush;
+      TpchDbGenerator(scale_factor, 100'000).generate_and_store();
+      std::cout << "done." << std::endl;
+
+      // Tables are dictionary-encoded for the following benchmarks.
+      const auto default_encoding = EncodingType::Dictionary;
+      std::cout << "Encoding tables ... " << std::flush;
+      ChunkEncoder::encode_all_chunks(sm.get_table("lineitem"),
+                                      std::vector<SegmentEncodingSpec>(sm.get_table("lineitem")->column_count(),
+                                                                       SegmentEncodingSpec(default_encoding)));
+      ChunkEncoder::encode_all_chunks(sm.get_table("orders"),
+                                      std::vector<SegmentEncodingSpec>(sm.get_table("orders")->column_count(),
+                                                                       SegmentEncodingSpec(default_encoding)));
+      std::cout << "done." << std::endl;
+    }
+
+    auto lineitem_tab = sm.get_table("lineitem");
+    auto orders_tab = sm.get_table("orders");
+
+    _lineitem_wrapper = std::make_shared<TableWrapper>(lineitem_tab);
+    _lineitem_wrapper->execute();
+
+    _orders_wrapper = std::make_shared<TableWrapper>(orders_tab);
```

Comment:
In case we start using more than two tables, this could be moved to a method that does this for all tables.